pkg/scheduler/cache/event_handlers.go

```diff
@@ -321,10 +322,82 @@ func (sc *SchedulerCache) deleteNode(node *v1.Node) error {
 	}
 
 	delete(sc.Nodes, node.Name)
+	// the deletion of this node should stop the synchronization for this node
+	if sc.notReadyNodes != nil {
+		sc.notReadyNodes.Forget(node.Name)
+	}
 
 	return nil
 }
 
+// Assumes that lock is already acquired.
+func (sc *SchedulerCache) syncNode(nodeInfo *schedulingapi.NodeInfo) error {
+	newNode, err := sc.getNode(nodeInfo.Name)
+	if err != nil || newNode == nil {
+		return err
+	}
+	tasks, err := sc.getTasksOnNode(nodeInfo.Name)
+	if err != nil {
+		return err
+	}
+	nodeInfo.State = schedulingapi.NodeState{Phase: schedulingapi.Ready}
+	nodeInfo.SetNode(newNode)
+	// remove unwanted tasks
+	for taskID, task := range nodeInfo.Tasks {
+		if _, exists := tasks[taskID]; exists {
+			continue
+		}
+		if err := sc.deleteTask(task); err != nil {
```

Comment:
IMO, maybe we can save the errors and continue to delete the following unwanted tasks when fails to delete some task. At the end of loop, throw all the errors to the caller. That may be more reasonable. In this way, we can ensure the most of the synchronization.

Reply:
"At the end of loop, throw all the errors to the caller" may result in big change of current schedule cycle, so i feel that the current fix may  have less impact on the current logic